lib/ansible/plugins/inventory/now.py

```diff
@@ -0,0 +1,202 @@
+            description: The ServiceNow instance URI. The URI should be the fully-qualified domain name, e.g. 'your-instance.servicenow.com'.
+            type: string
+            required: True
+            env:
+                - name: SN_INSTANCE
+        username:
+            description: The ServiceNow user acount, it should have rights to read cmdb_ci_server (default), or table specified by SN_TABLE
+            type: string
+            required: True
+            env:
+                - name: SN_USERNAME
+        password:
+            description: The ServiceNow instance user password.
+            type: string
+            secret: true
+            env:
+                - name: SN_PASSWORD
+        table:
+            description: The ServiceNow table to query
+            type: string
+            default: cmdb_ci_server
+        fields:
+            description: Comma seperated string providing additional table columns to add as host vars to each inventory host.
+            type: list
+            default: []
+        sn_groups:
+            description: Comma seperated string providing additional table columns to use as groups. Groups can overlap with fields
+            type: list
+            default: []
+        selection_order:
+            description: Comma seperated string providing ability to define selection preference order.
+            type: list
+            default: 'host_name,fqdn,ip_address'
+        filter_results:
+            description: Filter results with sysparm_query encoded query string syntax. Complete list of operators available for filters and queries.
+            type: string
+            default: ''
+        proxy:
+            description: Proxy server to use for requests to ServiceNow.
+            type: string
+            default: ''
+'''
+
+EXAMPLES = r'''
+plugin: now
+instance: demo.service-now.com
+username: admin
+password: password
+'''
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, to_safe_group_name, Cacheable
+from ansible.errors import AnsibleError, AnsibleParserError
+try:
+    import requests
+except ImportError:
+    raise AnsibleError('This script requires python-requests')
+import sys
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'now'
+
+    def invoke(self, verb, path, data):
+        auth = requests.auth.HTTPBasicAuth(self.get_option('username'),
+                                           self.get_option('password'))
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        proxy = self.get_option('proxy')
+
+        # build url
+        self.url = "https://%s/%s" % (self.get_option('instance'), path)
+        url = self.url
+
+        try:
+            results = self._cache[self.cache_key][self.url]
+        except KeyError:
+            self.update_cache = True
+
+        if not self.use_cache or self.url not in self._cache.get(
```

Comment:
You could make this easier to read by setting results = '' before line 99 and then just do `if not results:` here.